Can't install packages with npm@7  and Volta 0.9

Hi,

I just upgraded to the latest 0.9 and I can't install packages anymore 😞 

```
volta install netlify-cli
error: Could not install package 'netlify-cli'

Please confirm the package is valid and run with `--verbose` for more diagnostics.
[verbose] Installing netlify-cli with command: "npm" "install" "--global" "--loglevel=warn" "--no-update-notifier" "--no-audit" "netlify-cli"
[verbose] [install stderr]
npm ERR! code ENOENT
npm ERR! syscall lstat
npm ERR! path /Users/frank/.volta/tmp/.tmpO1Zq9R/lib
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lstat '/Users/frank/.volta/tmp/.tmpO1Zq9R/lib'
npm ERR! enoent This is related to npm not being able to find a file.
```

I get the same error when i run `npm install --global`.

It was working perfectly before, is there some node/npm config to check here?

```
which node
/Users/frank/.volta/bin/node

```

Thanks

Sure.

```
volta install npm@7
success: installed and set npm@7.0.3 as default

volta install yarn
success: installed and set yarn@1.22.10 as default

VOLTA_LOGLEVEL=debug RUST_BACKTRACE=full volta install netlify-cli
[verbose] Acquiring lock on Volta directory: /Users/frank/.volta/volta.lock
[verbose] Found default configuration at '/Users/frank/.volta/tools/user/platform.json'
[verbose] node@12.19.0 has already been fetched, skipping download
[verbose] npm@7.0.3 has already been fetched, skipping download
[verbose] yarn@1.22.10 has already been fetched, skipping download
[verbose] Installing netlify-cli with command: "npm" "install" "--global" "--loglevel=warn" "--no-update-notifier" "--no-audit" "netlify-cli"
[verbose] [install stderr]
npm ERR! code ENOENT
npm ERR! syscall lstat
npm ERR! path /Users/frank/.volta/tmp/.tmpb9aSag/lib
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lstat '/Users/frank/.volta/tmp/.tmpb9aSag/lib'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/frank/.npm/_logs/2020-10-22T14_14_19_169Z-debug.log

[verbose] [install stdout]

[verbose] Unlocking Volta Directory
error: Could not install package 'netlify-cli'

Please confirm the package is valid and run with `--verbose` for more diagnostics.
[verbose] No custom hooks found
```

--- a/crates/volta-core/src/tool/package/mod.rs
+++ b/crates/volta-core/src/tool/package/mod.rs
@@ -34,7 +34,8 @@ pub struct Package {
 
 impl Package {
     pub fn new(name: String, version: VersionSpec) -> Fallible<Self> {
-        let staging = create_staging_dir()?;
+        let staging = setup_staging_directory(PackageManager::Npm)?;
+
         Ok(Package {
             name,
             version,
@@ -124,7 +125,8 @@ pub struct DirectInstall {
 
 impl DirectInstall {
     pub fn new(manager: PackageManager) -> Fallible<Self> {
-        let staging = create_staging_dir()?;
+        let staging = setup_staging_directory(manager)?;
+
         Ok(DirectInstall { staging, manager })
     }
 
@@ -147,6 +149,22 @@ impl DirectInstall {
 
         Ok(())
     }
+}
+
+/// Create the temporary staging directory we will use to install and ensure expected
+/// subdirectories exist within it
+fn setup_staging_directory(manager: PackageManager) -> Fallible<TempDir> {
+    let staging = create_staging_dir()?;
+
+    let source_dir = manager.source_dir(staging.path().to_owned());
+    ensure_containing_dir_exists(&source_dir)
+        .with_context(|| ErrorKind::ContainingDirError { path: source_dir })?;
+
+    let binary_dir = manager.binary_dir(staging.path().to_owned());
+    ensure_containing_dir_exists(&binary_dir)
+        .with_context(|| ErrorKind::ContainingDirError { path: binary_dir })?;
+
+    Ok(staging)
 }
 
 fn persist_install<V>(package_name: &str, package_version: V, staging_dir: &Path) -> Fallible<()>

